engine/console/game.go

```diff
@@ -149,6 +149,43 @@ func (game *Game) handleGameEvent(event interface{}) {
 		}
 	case core.NewFood:
 		game.food = append(game.food, event.Pos)
+	case core.PlayerMove:
+		for id, dir := range event.Moves {
+			prevHead := game.snakes[id].head
+			// move snake's head
+			if dir == core.Up {
+				game.snakes[id].head = core.Coord{X: prevHead.X, Y: prevHead.Y - 1}
+			} else if dir == core.Left {
+				game.snakes[id].head = core.Coord{X: prevHead.X - 1, Y: prevHead.Y}
+			} else if dir == core.Right {
+				game.snakes[id].head = core.Coord{X: prevHead.X + 1, Y: prevHead.Y}
+			} else if dir == core.Down {
+				game.snakes[id].head = core.Coord{X: prevHead.X, Y: prevHead.Y + 1}
+			} else {
+				panic("the value of direction is unknown")
+			}
```

Comment:
A switch would look much cleaner. Additionally, you can just mutate the field in place, like
```
switch dir {
case core.Up:
    game.snakes[id].head.Y--
}
```